src/System.cc

```diff
@@ -533,19 +533,19 @@ void System::Shutdown()
     }*/
 
     // Wait until all thread have effectively stopped
-    /*while(!mpLocalMapper->isFinished() || !mpLoopCloser->isFinished() || mpLoopCloser->isRunningGBA())
+    while(!mpLocalMapper->isFinished() || !mpLoopCloser->isFinished() || mpLoopCloser->isRunningGBA())
```

Comment:
uncommented this initially to see if it fixed segfaults at the end of a run. Still see the mapper/loopcloser is not finished logs during Shutdown so leaving them in

Reply:
Is the purpose of this to allow more time for mapping / loop closing to finish? Also will rdk end up just force killing orbslam due to the extra five seconds?

Reply:
It’s only a 5 millisecond delay, not 5 seconds. Yeah the purpose is to let the other threads orbslam uses to finish. Fine with removing this if we don’t think it is necessary

Reply:
Oh, got it, then it shouldn't cause a force kill. Does this actually help the mapper/loopcloser finish?

Reply:
Usually i only see the print statements for each once, but occasionally if orbslam shutsdown mid run then the local mapping thread will take some extra time to finish.